### PR TITLE
site-docker: make it more readable

### PR DESCRIPTION
**Pull request by guits**
These conditions introduced by d981c6bd2 were insane.
This should be a bit easier to read.

Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

File: site-docker.yml.sample [hunk 1]
```diff
@@ -50,16 +50,16 @@
   roles:
     - { role: ceph-defaults,
         tags: [with_pkg, fetch_container_image],
-        when: "(((containerized_deployment | bool) and not (is_atomic | bool) and not (inventory_hostname in groups.get('clients', []))) or ((inventory_hostname == groups.get('clients', [''])|first) and (containerized_deployment | bool) and not (is_atomic | bool)))" }
+        when: "(containerized_deployment | bool) and not (is_atomic | bool) and (not (inventory_hostname in groups.get('clients', [])) or (inventory_hostname == groups.get('clients', [''])|first))" }
```
Comment (samdoran) [inline site-docker.yml.sample]: These can also be listed out to increase legibility. All list elements in the `when` field get joined with `and`.


```yaml
- role: ceph-defaults
  tags: [with_pkg, fetch_container_image]
  when:
    - containerized_deployment | bool
    - not (is_atomic | bool)
    - (not (inventory_hostname in groups.get('clients', [])) or (inventory_hostname == groups.get('clients', [''])|first))
```

Comment (guits) [reply site-docker.yml.sample]: @samdoran thanks!

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by leseb** [comment]
@guits luminous-ansible2.4-docker_cluster_collocation is failing, did you rebase on top of your mds patch?

**Comment by guits** [comment]
@leseb I thought I had but apparently no

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-bluestore_osds_container

**Comment by guits** [comment]
jenkins test this please

**Pull request merged**